# DSCI 100 Individual Planning Report

## Data Description
Firstly, we load required libraries and the data in.

In [4]:
# Load dataset
library(tidyverse)
library(repr)

In [6]:
# Load data
players <- read_csv("../data/players.csv")
sessions <- read_csv("../data/sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Now that we have the data loaded in, we can describe it and generate summary statistics:

In [2]:
summary(players)
summary(sessions)

nrow(players)
nrow(sessions)
ncol(players)
ncol(sessions)

players |> distinct(experience)

ERROR: Error in eval(expr, envir, enclos): object 'players' not found


### Summary of Data Description

The summary statistics and data from the above cell are summarized here:

For the **player dataset**:
- Number of variables: **7**
- Number of observations: **196**

**Player variables:**
- **experience:** A categorical variable of type character; one of "Beginner", "Amateur", "Regular", "Veteran", "Pro". Refers to the player's surveyed/self-declared experience with Minecraft. Collected by survey.
- **subscribe:** A categorical variable of type logical (``true`` or ``false``). Refers to whether player is subscribed to a game-related newsletter or not. Collected by survey.
- **hashedEmail:** A categorical variable of type character representing the player's hashed email. Presumably collected by signup.
- **played_hours:** A quantitative variable of type double representing the amount of hours they played on the server, collected by playtime logs.
    - Summary statitistics:
| Min | Q1  | Median | Mean | Q3 | Max | NA's|
|-----|-----|--------|------|----|-----|-----|
| 0.00|0.00|0.10|5.85|0.60|223.10|0|
- **name:** A categorical variable of type character representing the player's name, collected by survey.
- **gender:** A categorical variable of type character representing the player's self-identified gender, collected by survey.
- **age:** A quantitative variable of type double representing the player's age, collected by survey.
    - Summary statistics:
| Min | Q1  | Median | Mean | Q3 | Max | NA's|
|-----|-----|--------|------|----|-----|-----|
| 9.00|17.00|19.00|21.14|22.75|58.00|2|

For the **sessions dataset:**
- Number of variables: **5**
- Number of observations: **1535**

**Session variables:**
- **hashedEmail:** A categorical variable of type character representing the session's players' email, collected the same as player email.
- **start_time:** A categorical/ordinal variable of type character representing the session start time (dd/mm/yyyy, 24 hour time), collected by server logs.
- **end_time:** A categorical/ordinal variable of type character representing the session end time (dd/mm/yyyy, 24 hour time), collected by server logs.
- **original_start_time:** A quantitative variable of type double representing session's Unix start time (ms since Jan 1, 1970 at 00:00:00 UTC), collected by server logs.
    - Summary statistics:
| Min | Q1  | Median | Mean | Q3 | Max | NA's|
|-----|-----|--------|------|----|-----|-----|
| 1.71e+12|1.72e+12|1.72e+12|1.72e+12|1.72e+12|1.73e+12|0|
- **original_end_time:** A quantitative variable of type double representing session's Unix end time (ms since Jan 1, 1970 at 00:00:00 UTC), collected by server logs.
    - Summary statistics:
| Min | Q1  | Median | Mean | Q3 | Max | NA's|
|-----|-----|--------|------|----|-----|-----|
|1.71e+12|1.72e+12|1.72e+12|1.72e+12|1.72e+12|1.72e+12|2|


### Issues with data

Potential issues with this data include:
- Low player count (196) might not be enough to generate meaningful conclusions
- A few NA variables in age, original_end_time
- Summary statistics for original_start_time and original_end_time are not too specific as they are rounded to 2 decimal places